packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -332,8 +332,11 @@ async function prepareAndExecuteChangeSet(
     await cfn.executeChangeSet({ StackName: deployName, ChangeSetName: changeSetName, ...disableRollback }).promise();
 
     // eslint-disable-next-line max-len
+    const changeSetLength: number = (changeSetDescription.Changes ?? []).length;
     const monitor = options.quiet ? undefined : StackActivityMonitor.withDefaultPrinter(cfn, deployName, stackArtifact, {
-      resourcesTotal: (changeSetDescription.Changes ?? []).length,
+      // +2 for the COMPLETE event emitted from updates.
```

Comment:
It's confusing that this comment doesn't seem to match the code. 

It's probably better to remove the `+1` elsewhere and move the `+1`/`+2` here completely to match the comment.

Also, I don't quite understand why it's `+1` and `+2`. Naively, I would say we're waiting for either a stack `CREATE_COMPLETE` or a stack `UPDATE_COMPLETE`, which would both imply a `+1`.

What is the additional `+1` we're looking for in the case of updates?

Reply:
I have updated the comments in the code to reflect the functionality of the code.

I have kept the `+1` elsewhere as this elminates the need for repeated code as the class is used in multiple places and the `+1` is part of the constructor for the class.

The additional `+1` is needed as update events emit an extra event compared to creates, which hasn't been accounted for yet.